ledger-api/grpc-definitions/com/daml/ledger/api/v1/connected_domains_service.proto

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v1;
+
+option java_outer_classname = "ConnectedDomainsServiceOuterClass";
+option java_package = "com.daml.ledger.api.v1";
+option csharp_namespace = "Com.Daml.Ledger.Api.V1";
+
+// Allows clients to get state of connected domains from the ledger.
+service ConnectedDomainsService {
+  // Get the list of connected domains.
+  rpc GetConnectedDomains (GetConnectedDomainsRequest) returns (GetConnectedDomainsResponse);
```

Comment:
should we already support an offset in here? We don't strictly need it from our side but we will need offset-based topology information once we get to multi-hosted parties.

Reply:
AFAIK that is out of scope for the testnet.
TBH I see a hard time providing here an offset without implementing this, so I would go without for the testnet, and improve for 3.0
@rgugliel-da how do you see it?

Reply:
We have two different things:

- Connected domains (current): the list of domains the participant is currently connected to (really, connected). I don't think we have history there, so only "current" is what we can provide.
- Domains on which the participant is hosted: this can be grabbed from topology and we could imagine an offset (with some conversion since currently, we query topology snapshots per timestamp and not per offset). But that's doable.